src/App.tsx

```diff
@@ -1,26 +1,192 @@
+
+  const handleAddTodo = useCallback(
+    (
+      { title, userId, completed }: Omit<Todo, 'id'>,
+      setTitle: Dispatch<SetStateAction<string>>,
+    ) => {
+      setIsAddingTodo(true);
+      const todo = {
+        id: 0,
+        title,
+        userId,
+        completed,
+      };
+
+      setTempTodo(todo);
+
+      createTodo({ title, userId, completed })
+        .then(newTodo => {
+          setTodos(currentTodos => [...currentTodos, newTodo]);
+          setTitle('');
+        })
+        .catch(() => {
+          setErrorMessage(Errors.ADDING_TODO);
+          setTitle(title);
+        })
+        .finally(() => {
+          setIsAddingTodo(false);
+          setTempTodo(null);
+        });
+    },
+    [],
+  );
+
+  const handleDeleteTodo = useCallback((todoId: number) => {
+    setLoadingTodoIds(currentLoadingTodoIds => [
+      ...currentLoadingTodoIds,
+      todoId,
+    ]);
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+      })
+      .catch(() => setErrorMessage(Errors.DELETE_TODO))
+      .finally(() => {
+        setLoadingTodoIds([]);
+      });
+  }, []);
+
+  const handleUpdateTodo = useCallback(
+    (
+      todoToUpdate: Todo,
+      isEditing?: boolean,
+      setIsEditing?: Dispatch<SetStateAction<boolean>>,
+    ) => {
+      setLoadingTodoIds(currentLoadingTodosIds => [
+        ...currentLoadingTodosIds,
+        todoToUpdate.id,
+      ]);
+
+      updateTodo(todoToUpdate)
+        .then(updatedTodo => {
+          setTodos(currentTodos => {
+            const newTodos = [...currentTodos];
+            const index = newTodos.findIndex(
+              todo => todo.id === updatedTodo.id,
+            );
+
+            newTodos.splice(index, 1, updatedTodo);
+
+            return newTodos;
+          });
+
+          if (isEditing) {
+            setIsEditing!(false);
+          }
+        })
+        .catch(() => setErrorMessage(Errors.UPDATE_TODO))
+        .finally(() => setLoadingTodoIds([]));
+    },
+    [],
+  );
 
-export const App: React.FC = () => {
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">
-          React Todo App - Add and Delete
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          todos={todos}
+          setErrorMessage={setErrorMessage}
+          handleAddTodo={handleAddTodo}
+          isAddingTodo={isAddingTodo}
+          errorMessage={errorMessage}
+          handleUpdateTodo={handleUpdateTodo}
```

Comment:
Use handle prefix for event handlers in the parent and on prefix in the child. 
This maintains clarity and indicates the function's source.